### PR TITLE
Allow to specify test by signature

### PR DESCRIPTION
**Pull request by iFrostizz**
Closes: https://github.com/runtimeverification/evm-semantics/issues/1870

The issue with specifying tests by method name is that in the case of overloaded functions, they can be ambiguous (e.g having a function whose signature is `func()` and another whose signature is `func(uint8)`).
This PR should not break the past behaviour of entering a method name for the test, ~~but if it matches more than one test, then we raise an exception asking the user to be more precise until there is only 1 matching test.~~
Edit: we now use foundry-like `--match-test` flag with regex. If `>` 1 test matches, we run them all at once.

This is probably breaking a lot of other utils relying on method by name (not only foundry-prove)

Checklist of foundry-* commands that are not broken

- [x] kompile
- [x] prove
- [x] show
- [x] to-dot (getting another error related to  an internal)
- [x] list
- [x] remove-node
- [x] simplify-node
- [x] step-node
- [x] section-edge

### CODE REVIEW

**Comment by ehildenb** [comment]
Back out the K submodule change please, and merge to latest master.

File: kevm-pyk/src/kevm_pyk/foundry.py [hunk 1]
```diff
@@ -684,8 +744,9 @@ def foundry_list(foundry_root: Path) -> list[str]:
 def foundry_remove_node(foundry_root: Path, test: str, node: NodeIdLike) -> None:
     foundry = Foundry(foundry_root)
     apr_proofs_dir = foundry.out / 'apr_proofs'
-    contract_name, test_name = test.split('.')
-    proof_digest = foundry.proof_digest(contract_name, test_name)
+    contract_name = test.split('.')[0]
+    test_sig = foundry.matching_sig(test).split('.')[1]
+    proof_digest = foundry.proof_digest(contract_name, test_sig)
```
Comment (anvacaru) [inline kevm-pyk/src/kevm_pyk/foundry.py:749]: This logic seems to be duplicated a few times. Can it be extracted in a new function?

Comment (iFrostizz) [reply kevm-pyk/src/kevm_pyk/foundry.py:749]: Thanks!

File: kevm-pyk/src/kevm_pyk/foundry.py [hunk 2]
```diff
@@ -684,8 +746,8 @@ def foundry_list(foundry_root: Path) -> list[str]:
 def foundry_remove_node(foundry_root: Path, test: str, node: NodeIdLike) -> None:
     foundry = Foundry(foundry_root)
     apr_proofs_dir = foundry.out / 'apr_proofs'
-    contract_name, test_name = test.split('.')
-    proof_digest = foundry.proof_digest(contract_name, test_name)
+    (contract_name, test_sig) = foundry.unique_sig(test)
+    proof_digest = foundry.proof_digest(contract_name, test_sig)
```
Comment (anvacaru) [inline kevm-pyk/src/kevm_pyk/foundry.py:750]: This logic seems duplicated across some `foundry_*` commands. Can it be extracted?

**Comment by anvacaru** [comment]
Looks good to me, but I request another review on this one.

File: kevm-pyk/src/kevm_pyk/foundry.py (hunk 2 above)
Comment (iFrostizz) [reply kevm-pyk/src/kevm_pyk/foundry.py]: Thanks! Done

**Review by anvacaru — APPROVED** [review]

**Pull request merged**